Increased number of quality control checks and analysis of results

s-good opened the pull request:
Added two basic quality control checks to allow the combining of tests to be observed. Added new benchmarking code to analyse the results and print out and plot them. This is similar to the functionality described in #22.

NOTE - The two range checks have been changed so that they reject some profiles and are not currently working correctly.

To do - Add the output to the log files rather than print to screen; determine how to cope when the number of tests increase and the number of combinations massively increases.

BillMills commented on util/benchmarks.py:
```diff
@@ -0,0 +1,91 @@
+import combineTests as combinatorics
+import matplotlib.pyplot as plt
+import numpy as np
+import sag_utilities as sag
```
This import doesn't appear to be used (and pip doesn't know what `sag_utilities` is).

BillMills commented:
Asides from the possibly spurious `sag_utilities` pointed out above, this looks good by eye. Couple things to add to the todo list:

 - Docs to help interpret the figures.
 - Testing in `util/`

But, both those things can get added after merging this PR. Can we safely take out `sag_utilities`?

s-good commented:
Sorry, the sag_utilities is a local module that I was using for debugging. It can safely be removed.

I've added an extra commit where the import is removed. I've also tidied up the output from the plot_roc function as even with four quality control checks there are too many combinations to list them all. I've added some brief extra comments explaining the plot.
